Source/JavaScriptCore/runtime/Options.cpp

```diff
@@ -628,24 +628,35 @@ void Options::setAllJITCodeValidations(bool value)
     Options::useJITAsserts() = value;
 }
 
-static inline void disableAllWasmOptions()
+static inline void disableAllWasmJITOptions()
 {
-    Options::useWasm() = false;
-    Options::useWasmIPInt() = false;
-    Options::useWasmLLInt() = false;
+    Options::useLLInt() = true;
+    Options::useWasmJIT() = false;
     Options::useBBQJIT() = false;
     Options::useOMGJIT() = false;
+
+    Options::useWasmSIMD() = false;
+
     Options::dumpWasmDisassembly() = false;
     Options::dumpBBQDisassembly() = false;
     Options::dumpOMGDisassembly() = false;
+}
+
+static inline void disableAllWasmOptions()
+{
+    disableAllWasmJITOptions();
+
+    Options::useWasm() = false;
```

Comment:
This is the crux of the issue; we absolutely must not force-disable Wasm itself given where this is being called from.